### PR TITLE
AV-565, AV-1234, AV-1233: Update Dependencies

### PR DESCRIPTION
**Pull request by jakecoble**
Stories:

- [AV-565](https://audioverse7.atlassian.net/jira/software/projects/AV/boards/3?assignee=712020:e12535da-4ed4-4700-9b40-8821b41cbf9e&selectedIssue=AV-565)
- [AV-1234](https://audioverse7.atlassian.net/jira/software/projects/AV/boards/3?assignee=712020:e12535da-4ed4-4700-9b40-8821b41cbf9e&selectedIssue=AV-1234)
- [AV-1233](https://audioverse7.atlassian.net/jira/software/projects/AV/boards/3?assignee=712020:e12535da-4ed4-4700-9b40-8821b41cbf9e&selectedIssue=AV-1233)

[AV-565]: https://audioverse7.atlassian.net/browse/AV-565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AV-1234]: https://audioverse7.atlassian.net/browse/AV-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AV-1233]: https://audioverse7.atlassian.net/browse/AV-1233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by narthur — CHANGES_REQUESTED** [review]
Good work!

File: src/components/molecules/searchBar.tsx [hunk 1]
```diff
@@ -91,10 +91,10 @@ export default function SearchBar({
 						description: 'search bar label',
 					})}
 					onBlur={() => {
-						setIsFocused(false),
-							lastTerm != term && term != ''
-								? (analytics.track('Search', { term }), setLastTerm(term))
-								: '';
+						setIsFocused(false);
+						return lastTerm != term && term != ''
```
[the remark this answers is not in the document]
Comment (jakecoble) [reply src/components/molecules/searchBar.tsx:95]: Failed to notice that this lambda doesn't return the result of its last expression.

File: src/lib/useNavigationItems.tsx [hunk 2]
```diff
@@ -25,100 +25,126 @@ const opts = {
...
 
 const iconMap = {
-	icon_search: dynamic(() => import('~public/img/icons/icon-search.svg'), opts),
-	icon_books: dynamic(() => import('~public/img/icons/icon-books.svg'), opts),
-	fa_bookmark: dynamic(() => import('~public/img/icons/fa-bookmark.svg'), opts),
-	fa_stream: dynamic(() => import('~public/img/icons/fa-stream.svg'), opts),
-	fa_history: dynamic(() => import('~public/img/icons/fa-history.svg'), opts),
-	icon_bible: dynamic(() => import('~public/img/icons/icon-bible.svg'), opts),
+	icon_search: dynamic(() => import('~public/img/icons/icon-search.svg'), {
+		...opts,
```
[the remark this answers is not in the document]
Comment (jakecoble) [reply src/lib/useNavigationItems.tsx]: Found the reason for the spreading.

https://github.com/audioverse-org/audioverse-next/actions/runs/10668784892/job/29569272384?pr=576#step:7:23
https://nextjs.org/docs/messages/invalid-dynamic-options-type

**Comment by narthur** [comment]
@jakecoble Looks like the accessibility scan found a bunch of new instances of this error:

WCAG2AA.Principle2.Guideline2_4.2_4_1.H64.1

https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/

**Review by narthur — APPROVED** [review]
> Reviewed 336 of 336 files at r1.
> *[Reviewable](https://reviewable.io/reviews/audioverse-org/audioverse-next/576)* status: 333 of 336 files reviewed, 13 unresolved discussions (waiting on @alangumer and @jakecoble)

___
*[`.husky/pre-commit` line 3 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7y67N3Baj2Aooob18G:-O7y67N3Baj2Aooob18H:b-aso6t9) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/b16dbb14c377169cdbfd71bebb4f6c3e56160f23/.husky/pre-commit#L3)):*
> ```
> #!/usr/bin/env sh
> . "$(dirname -- "$0")/_/husky.sh"
> ```

Did you confirm that the hook still works after this change?
___
*[`src/components/molecules/navItem.tsx` line 43 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7y9iOJEmtxy6x15pPc:-O7y9iOJEmtxy6x15pPd:bbpcl30) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/e954d23fff44d3b6852f370686886da537818f80/src/components/molecules/navItem.tsx#L43)):*
> ```XML
> 					onClick={() => {
> 						analytics.track('Menu click', { label });
> 						return setSubmenu(key);
> ```

I don't think this needs to be returned.
___
*[`src/components/organisms/cardSlider/index.spec.tsx` line 17 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7yDbkM79fBRXSGBwDW:-O7yDbkM79fBRXSGBwDX:bdk7ku9) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/e954d23fff44d3b6852f370686886da537818f80/src/components/organisms/cardSlider/index.spec.tsx#L17)):*
> ```XML
> 	previous: 'previous',
> 	next: 'next',
> 	items: [undefined],
> ```

Why is this needed?
___
*[`src/components/organisms/cardSlider/section/continueListening.tsx` line 33 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7yDpKi0k2fv7-dqrhR:-O7yDpKi0k2fv7-dqrhS:b-graw8d) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/e954d23fff44d3b6852f370686886da537818f80/src/components/organisms/cardSlider/section/continueListening.tsx#L33)):*
> ```XML
> 			selectNodes={(p) =>
> 				(
> 					p as GetSectionContinueListeningQuery
> ```

Ideally this would be inferred automatically from the other passed attributes. But I understand that can be tricky.
___
*[`src/components/templates/andPlaybackContext.tsx` line 464 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7yEVB-7-3h8tr4N0Xu:-O7yEVB-7-3h8tr4N0Xv:b-1anzel) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/b16dbb14c377169cdbfd71bebb4f6c3e56160f23/src/components/templates/andPlaybackContext.tsx#L464)):*
> ```XML
> 			};
> 
> 			const options: VideoJs.VideoJsPlayerOptions = {
> ```

It would be nice if we could figure out a way to keep this
___
*[`src/containers/base.spec.tsx` line 50 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7yEod_CNIx5TzOSoa2:-O7yEod_CNIx5TzOSoa3:b-o6nzmq) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/e954d23fff44d3b6852f370686886da537818f80/src/containers/base.spec.tsx#L50)):*
> ```XML
> 				const { data: myQuery } = useQuery({
> 					queryKey: ['myQuery'],
> 					...jest.fn(),
> ```

This looks weird to me. Is spreading `jest.fn()` the right way to do this?
___
*[`src/lib/usePlaybackSession.tsx` line 87 at r1](https://reviewable.io/reviews/audioverse-org/audioverse-next/576#-O7yGOKVEVrBxPexImgi:-O7yGOKVEVrBxPexImgj:b-va7r8r) ([raw file](https://github.com/audioverse-org/audioverse-next/blob/e954d23fff44d3b6852f370686886da537818f80/src/lib/usePlaybackSession.tsx#L87)):*
> ```XML
> 		},
> 		{
> 			queryKey: ['playback-progress', recording?.id || 0],
> ```

Why are you manually setting the query key here? Shouldn't that be handled by the generated code?

File: src/components/organisms/cardSlider/section/continueListening.tsx [hunk 3]
```diff
@@ -26,7 +29,9 @@ export default function ContinueListening(): JSX.Element {
 				defaultMessage: 'Next continue listening',
 			})}
 			selectNodes={(p) =>
-				p?.me?.user.continueListening.nodes?.map((n) => n.recording)
+				(
+					p as GetSectionContinueListeningQuery
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/continueListening.tsx:33]: The type assertion is not necessary, remove it.

File: src/components/organisms/cardSlider/section/continueListening.tsx [hunk 4]
```diff
@@ -26,7 +29,9 @@ export default function ContinueListening(): JSX.Element {
...
+				)?.me?.user.continueListening.nodes?.map((n) => n.recording)
 			}
 			selectPageInfo={(p) => p?.me?.user.continueListening.pageInfo}
 			Card={(p: { node: CardRecordingFragment }) => (
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/continueListening.tsx:37]: Same here.

File: src/components/organisms/cardSlider/section/trendingMusic.tsx [hunk 5]
```diff
@@ -29,7 +32,11 @@ export default function TrendingMusic(): JSX.Element {
 				defaultMessage: 'Next trending scripture songs',
 			})}
 			seeAllUrl={root.lang(route).songs.albums.get()}
-			selectNodes={(p) => p?.trendingMusic.nodes?.map((n) => n.recording)}
+			selectNodes={(p) =>
+				(p as GetSectionTrendingMusicQuery)?.trendingMusic.nodes?.map(
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/trendingMusic.tsx:36]: The type assertion is not necessary, remove it.

File: src/components/organisms/cardSlider/section/trendingMusic.tsx [hunk 6]
```diff
@@ -29,7 +32,11 @@ export default function TrendingMusic(): JSX.Element {
...
+					(n) => n.recording,
+				)
+			}
 			Card={(p: { node: CardRecordingFragment }) => (
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/trendingMusic.tsx:40]: Same here.

File: src/components/organisms/cardSlider/section/trendingTeachings.tsx [hunk 7]
```diff
@@ -29,7 +32,11 @@ export default function TrendingTeachings(): JSX.Element {
 				defaultMessage: 'Next trending teachings',
 			})}
 			seeAllUrl={root.lang(route).teachings.trending.get()}
-			selectNodes={(p) => p?.trendingTeachings.nodes?.map((n) => n.recording)}
+			selectNodes={(p) =>
+				(p as GetSectionTrendingTeachingsQuery)?.trendingTeachings.nodes?.map(
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/trendingTeachings.tsx:36]: The type assertion is not necessary, remove it.

File: src/components/organisms/cardSlider/section/trendingTeachings.tsx [hunk 8]
```diff
@@ -29,7 +32,11 @@ export default function TrendingTeachings(): JSX.Element {
...
+					(n) => n.recording,
+				)
+			}
 			Card={(p: { node: CardRecordingFragment }) => (
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/trendingTeachings.tsx:40]: Same here.

File: src/components/organisms/cardSlider/section/index.tsx [hunk 9]
```diff
@@ -77,8 +77,7 @@ export default function Section<T extends GraphqlInfiniteQuery, N>({
 
 	const { data, fetchNextPage, isFetchingNextPage, hasNextPage } =
 		infiniteQuery(
-			'after',
-			{ language },
+			{ language, after: 'after' },
```
Comment (alangumer) [inline src/components/organisms/cardSlider/section/index.tsx:80]: Why after: 'after'?

File: src/containers/presenter/list/all.tsx [hunk 10]
```diff
@@ -15,19 +15,19 @@ export default function All(props: PresentersProps) {
 
 	const { data, hasNextPage, fetchNextPage, isFetching } =
 		useInfiniteGetPresenterListAllPageDataQuery(
-			'after',
 			{
 				language,
-				after: null,
+				after: 'after',
```
Comment (alangumer) [inline src/containers/presenter/list/all.tsx:20]: Same here.

File: src/containers/sponsor/list/all.tsx [hunk 11]
```diff
@@ -21,15 +21,15 @@ export default function AllSponsors(props: Props) {
 
 	const { data, hasNextPage, fetchNextPage, isLoading, isFetchingNextPage } =
 		useInfiniteGetSponsorListAllPageDataQuery(
-			'after',
-			{ language },
+			{ language, after: 'after' },
```
Comment (alangumer) [inline src/containers/sponsor/list/all.tsx:24]: Same here.

**Review by alangumer — CHANGES_REQUESTED** [review]
Great work! Just a couple of comments.

File: src/components/organisms/cardSlider/section/trendingMusic.tsx (hunk 6 above)
Comment (alangumer) [reply src/components/organisms/cardSlider/section/trendingMusic.tsx]: Never mind.

File: src/components/organisms/cardSlider/section/index.tsx
Comment (alangumer) [inline src/components/organisms/cardSlider/section/index.tsx:1]: It looks that for the type inference to work for selectPageInfo when the component is used, the type should be PageInfoSelector<InferGraphqlInfiniteQueryType<T>>.

**Review by alangumer — CHANGES_REQUESTED** [review]

**Review by alangumer — APPROVED** [review]

File: src/components/organisms/cardSlider/section/index.tsx
Comment (narthur) [reply src/components/organisms/cardSlider/section/index.tsx:1]: Resolved

**Pull request merged**